src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java

```diff
@@ -611,4 +623,33 @@ public boolean isDeconstructed()
     {
         return isDeconstructed;
     }
+
+    @NotNull
+    @Override
+    public <T extends IBuildingModuleView> Optional<T> getModuleView(final Class<T> clazz)
+    {
+        final List<T> list = getModuleViews(clazz);
+        if (list.isEmpty())
+        {
+            return Optional.empty();
+        }
+        return Optional.of(list.get(0));
```

Comment:
This can be made cheaper.
Right now you do a full loop when you call getModuleViews(...), it is cheaper to implement this seperately and do a findFirst(....)